pkg/cli/cli.go

```diff
@@ -983,14 +1001,99 @@ func printConnection(c *command, cvd RemoteCVDLocator, status ConnStatus) {
 	c.Printf("%s/%s: %s\n", cvd.Host, cvd.WebRTCDeviceID, state)
 }
 
-func buildAgentCmdArgs(flags *ConnectFlags, device string) []string {
-	args := []string{
-		ConnectionAgentCommandName,
-		device,
-	}
+func buildAgentCmdArgs(flags *ConnectFlags, device string, connectAgent string) []string {
+	args := []string{connectAgent, device}
 	return append(args, flags.AsArgs()...)
 }
 
+func runConnectionProxyAgentCommand(flags *ConnectFlags, c *command, args []string, opts *subCommandOpts) error {
+	if len(args) > 1 {
+		return fmt.Errorf("connection agent only supports a single device, received: %v", args)
+	}
+	if len(args) == 0 {
+		return errors.New("missing device")
+	}
+	device := args[0]
+	service, err := opts.ServiceBuilder(flags.CVDRemoteFlags, c.Command)
+	if err != nil {
+		return err
+	}
+	hosts, err := service.ListHosts()
+	if err != nil {
+		return fmt.Errorf("error listing hosts: %w", err)
+	}
+	var host *apiv1.HostInstance
+	for _, instance := range hosts.Items {
+		if instance.Name == flags.host {
+			host = instance
+		}
+	}
+	if host == nil {
+		return errors.New("failed to find host")
+	}
+	if host.Docker == nil {
+		return errors.New("instance type should be Docker")
+	}
+
+	controlDir := opts.InitialConfig.ConnectionControlDirExpanded()
+	cvdHosts, err := listCVDsSingleHost(service, controlDir, flags.host)
+	if err != nil {
+		return fmt.Errorf("error listing CVDs: %w", err)
+	}
+	var port string
+	for _, cvd := range cvdHosts[0].CVDs {
+		if device == cvd.WebRTCDeviceID {
+			_, port, err = net.SplitHostPort(cvd.ADBSerial)
+			if err != nil {
+				return fmt.Errorf("failed to parse port from ADB serial: %w", err)
+			}
+			break
+		}
+	}
+	if port == "" {
+		return errors.New("failed to find port")
+	}
+
+	var dialer proxy.Dialer
+	if flags.HTTPProxy != "" {
+		proxyUrl, err := url.Parse(flags.HTTPProxy)
+		if err != nil {
+			return fmt.Errorf("failed to parse proxy URL: %w", err)
+		}
+		if proxyUrl.Scheme != "socks5" {
+			return fmt.Errorf("scheme of proxy URL is not socks5. actual: %s", proxyUrl.Scheme)
+		}
+		dialer, err = proxy.SOCKS5("tcp", proxyUrl.Host, nil, nil)
+		if err != nil {
+			return fmt.Errorf("failed to create proxy dialer: %w", err)
+		}
+	} else {
+		dialer = proxy.Direct
+	}
+	remoteConn, err := dialer.Dial("tcp", net.JoinHostPort(host.Docker.IPAddress, port))
+	if err != nil {
+		return fmt.Errorf("failed to dial remote port: %w", err)
+	}
+	defer remoteConn.Close()
+
+	shortenHost := flags.host
+	if len(shortenHost) >= 8 {
+		shortenHost = shortenHost[:8]
+	}
+	socketName := strings.Join([]string{"proxy", shortenHost, device}, "_") + ".sock"
```

Comment:
let's move the logic to build to socket name into a helper function. use `fmt.Sprintf()`

Reply:
Done.